src/server/req_delete.c

```diff
@@ -834,6 +836,33 @@ req_deletejob2(struct batch_request *preq, job *pjob)
 	reply_send(preq);
 }
 
+/**
+ * @brief
+ * 		req_reservationOccurrenceEnd - service the PBS_BATCH_ResvOccurEnd Request
+ *
+ *		This request runs a hook script at the end of the reservation occurrence
+ *
+ * @param[in]	preq	- Job Request
+ *
+ */
+
+
+void req_reservationOccurrenceEnd(struct batch_request *preq)
+{
+	char hook_msg[HOOK_MSG_SIZE] = {0};
+
+        switch (process_hooks(preq, hook_msg, sizeof(hook_msg), pbs_python_set_interrupt)) {
+		case 0:	/* explicit reject */
+			free_br(preq);
```

Comment:
I don't think that you should be freeing up batch request here. You should either do a reply_ack on batch request in positive cases or reply_reject in negative cases. If you just free up batch request in the positive case then who will free this batch request when a hook rejects or no hook runs or when there is an internal failure.
Another thing is that if you do not reply back using the batch_request then I'm not sure who cleans up the deferred local work task from server.

I'd also suggest you run your tests with the server running under valgrind to check for leaks and other issues.